osquery/filesystem/filesystem.cpp

```diff
@@ -147,6 +148,120 @@ Status listFilesInDirectory(const boost::filesystem::path& path,
   }
 }
 
+Status listDirectoriesInDirectory(const boost::filesystem::path& path,
+                            std::vector<std::string>& results) {
+  try {
+    if (!boost::filesystem::exists(path)) {
+      return Status(1, "Directory not found");
+    }
+
+    if (!boost::filesystem::is_directory(path)) {
```

Comment:
https://github.com/facebook/osquery/blob/master/osquery/filesystem/filesystem.cpp#L159

Then move this out of a try.